python/tank/bootstrap/configuration_writer.py

```diff
@@ -119,76 +163,86 @@ def get_descriptor_metadata_file(self):
 
         """
-        config_backup_path = None
-        core_backup_path = None
 
-        # get backup root location
-        config_path = self._path.current_os
-        configuration_payload = os.path.join(config_path, "config")
-
-        # timestamp for rollback backup folders
-        timestamp = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
-
-        if os.path.exists(configuration_payload):
-
-            config_backup_root = os.path.join(config_path, "install", "config.backup")
-
-            # make sure we have a backup folder present
-            # sometimes, execution and rollback is so quick that several backup folders
-            # are created in a single second. In that case, append a suffix
-            config_backup_path = os.path.join(config_backup_root, timestamp)
-            counter = 0
-            while os.path.exists(config_backup_path):
-                # that backup path already exists. Try another one
-                counter += 1
-                config_backup_path = os.path.join(config_backup_root, "%s.%d" % (timestamp, counter))
-
-            # now that we have found a spot for our backup, make sure folder exists
-            # and then move the existing config *into* this folder.
-            filesystem.ensure_folder_exists(config_backup_path)
-
-            log.debug("Moving config %s -> %s" % (configuration_payload, config_backup_path))
-            backup_target_path = os.path.join(config_backup_path, os.path.basename(configuration_payload))
-            os.rename(configuration_payload, backup_target_path)
-            log.debug("Backup complete.")
-            config_backup_path = backup_target_path
-
-        # now back up the core API
-        core_payload = os.path.join(config_path, "install", "core")
-
-        if os.path.exists(core_payload):
-            core_backup_root = os.path.join(config_path, "install", "core.backup")
-            # should not be necessary but just in case.
-            filesystem.ensure_folder_exists(core_backup_root)
-
-            # make sure we have a backup folder present
-            # sometimes, execution and rollback is so quick that several backup folders
-            # are created in a single second. In that case, append a suffix
-            core_backup_path = os.path.join(core_backup_root, timestamp)
-            counter = 0
-            while os.path.exists(core_backup_path):
-                # that backup path already exists. Try another one
-                counter += 1
-                core_backup_path = os.path.join(core_backup_root, "%s.%d" % (timestamp, counter))
-
-            log.debug("Moving core %s -> %s" % (core_payload, core_backup_path))
-            os.rename(core_payload, core_backup_path)
-            log.debug("Backup complete.")
-            core_backup_path = core_backup_path
-
-        return (config_backup_path, core_backup_path)
+        # The rename guard's role is to keep track of every rename operation that has happened
```

Comment:
Don't let the huge diff fool you, the only thing that actually changed in this block is the indentation level because of the with statement.